plugins/scaffolder-backend/src/service/router.ts

```diff
@@ -232,6 +232,9 @@ export async function createRouter(
       }
 
       const result = await taskBroker.dispatch(taskSpec, {
+        ...req.body.secrets,
```

Comment:
I feel this and the other things really should get some validation immediately at the start. Time for some zod (some other day).